client/lib/purchases/index.js

```diff
@@ -481,7 +541,7 @@ function purchaseType( purchase ) {
 		return i18n.translate( 'Site Plan' );
 	}
 
-	if ( isDomainRegistration( purchase ) ) {
+	if ( isDomainRegistration( purchase ) || isJetpackProduct( purchase ) ) {
```

Comment:
This is causing some duplicate product names to appear on the purchase page itself:
![product-page](https://user-images.githubusercontent.com/235183/81866814-bfb3d980-953d-11ea-8efc-25536f5e2cf2.png)

And doesn't seem too useful in the modal dialog, since it duplicates there too:
![modal-dialog](https://user-images.githubusercontent.com/235183/81866816-bfb3d980-953d-11ea-9558-736170904819.png)

I think it might be best to just continue leaving this blank and then make the modal dialog smart enough to handle the possibility of an empty purchase type (e.g. by just displaying "Expires in two months..." in that case)?